emission/analysis/intake/segmentation/trip_segmentation_methods/dwell_segmentation_time_filter.py

```diff
@@ -57,127 +61,146 @@ def __init__(self, time_threshold, point_threshold, distance_threshold):
-                # So we reset_index upstream and use it here.
-                # We are going to use the last 8 points for now.
-                # TODO: Change this back to last 10 points once we normalize phone and this
-                last10Points_df = filtered_points_df.iloc[
-                    max(idx - self.point_threshold, curr_trip_start_point.idx):idx + 1
+            while trip_start_idx < len(loc_df):
+                logging.info(f"trip_start_idx = {trip_start_idx}")
+                # trim off dists of points that were before the trip_start_idx
+                recent_diffs_filtered = [
+                    diffs if row_idx - trip_start_idx > diffs.shape[1]
+                    else diffs[:, trip_start_idx - row_idx:]
+                    for row_idx, diffs in loc_df['recent_points_diffs'].items()
                 ]
-                distanceToLast = lambda row: pf.calDistance(ad.AttrDict(row), currPoint)
-                timeToLast = lambda row: currPoint.ts - ad.AttrDict(row).ts
-                last5MinsDistances = last5MinsPoints_df.apply(distanceToLast, axis=1)
-                logging.debug("last5MinsDistances = %s with length %d" % (last5MinsDistances.to_numpy(), len(last5MinsDistances)))
-                last10PointsDistances = last10Points_df.apply(distanceToLast, axis=1)
-                logging.debug("last10PointsDistances = %s with length %d, shape %s" % (
-                    last10PointsDistances.to_numpy(),
-                    len(last10PointsDistances),
-                    last10PointsDistances.shape
-                ))
-
-                # Fix for https://github.com/e-mission/e-mission-server/issues/348
-                last5MinTimes = last5MinsPoints_df.apply(timeToLast, axis=1)
-
-                logging.debug("len(last10PointsDistances) = %d, len(last5MinsDistances) = %d" %
-                            (len(last10PointsDistances), len(last5MinsDistances)))
-                logging.debug("last5MinTimes.max() = %s, time_threshold = %s" %
-                            (last5MinTimes.max() if len(last5MinTimes) > 0 else np.NaN, self.time_threshold))
-
-
-                with ect.Timer() as t_has_trip_ended:
-                    if self.has_trip_ended(prevPoint, currPoint, timeseries, last10PointsDistances, last5MinsDistances, last5MinTimes):
-                        (ended_before_this, last_trip_end_point) = self.get_last_trip_end_point(
-                            filtered_points_df,
-                            last10Points_df,
-                            last5MinsPoints_df
+
+                max_recent_dist_diffs = np.array(
+                    [np.inf
+                        # 'inf' means we can't make a decision because:
+                        # we don't have enough recent points (TODO the -2 weird, but necessary to match the current behavior)
+                        if diffs.shape[1] < self.point_threshold - 2
+                        # or the oldest point within the time filter is not old enough
```

Comment:
where are we doing `-2` now? Can you clarify as a response to this comment?

Reply:
It was `-1` below, but `last10Points` included the current point so we got one for free

https://github.com/e-mission/e-mission-server/blob/56b91a681e5d10fc07a4000431d9ddffd1867abc/emission/analysis/intake/segmentation/trip_segmentation_methods/dwell_segmentation_time_filter.py#L292

In the new version we have `diffs` which does not include the current point, so it's `-2`